src/Kiota.Builder/Writers/Ruby/CodeMethodWriter.cs

```diff
@@ -98,6 +103,14 @@ public override void WriteCodeElement(CodeMethod codeElement, LanguageWriter wri
             writer.IncreaseIndent();
             writer.WriteLine($"return @{codeElement.AccessedProperty?.Name?.ToSnakeCase()}");
         }
+        private void WriteIndexerBody(CodeMethod codeElement, LanguageWriter writer, string returnType) {
+            var prefix = string.Empty;
+            if(codeElement.ReturnType is CodeType xType && xType.TypeDefinition is CodeClass xCodeClass && xCodeClass?.Parent is CodeNamespace ns)
+                prefix = $"{ns.Name.NormalizeNameSpaceName("::")}::";
```

Comment:
you probably need to do that for enums as well (type definition can be a code enum)

Reply:
Would Enums ever need to write an IndexerBody?